src/main/java/com/lambdaworks/redis/protocol/AsyncCommand.java

```diff
@@ -102,6 +102,7 @@ public boolean completeExceptionally(Throwable ex) {
     @Override
     public boolean cancel(boolean mayInterruptIfRunning) {
         command.cancel();
+        latch.countDown();
```

Comment:
The latch should be released after invoking cancellation to avoid a state in which the latch is released but the `Future` is not canceled yet. I think wrapping the two cancel calls into a `try` and releasing the latch in `finally` should do the job.